cumulus/parachains/runtimes/constants/src/lib.rs

```diff
@@ -15,7 +15,159 @@
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
+pub use constants::*;
+pub use default_configs::*;
+
 #[cfg(feature = "rococo")]
 pub mod rococo;
 #[cfg(feature = "westend")]
 pub mod westend;
+
+pub mod default_configs;
+
+pub mod constants {
+	use frame_support::{
+		pallet_prelude::*,
+		parameter_types,
+		weights::{constants::WEIGHT_REF_TIME_PER_NANOS, RuntimeDbWeight, Weight},
+	};
+	use frame_system::limits::{BlockLength, BlockWeights};
+	use parachains_common::{
+		AVERAGE_ON_INITIALIZE_RATIO, MAXIMUM_BLOCK_WEIGHT, NORMAL_DISPATCH_RATIO,
+	};
+
+	parameter_types! {
+		/// By default, Substrate uses `RocksDB`, so this will be the weight used throughout
+		/// the runtime.
+		pub const RocksDbWeight: RuntimeDbWeight = RuntimeDbWeight {
+			read: 25_000 * WEIGHT_REF_TIME_PER_NANOS,
+			write: 100_000 * WEIGHT_REF_TIME_PER_NANOS,
+		};
+		/// `ParityDB` can be enabled with a feature flag, but is still experimental. These weights
+		/// are available for brave runtime engineers who may want to try this out as default.
+		pub const ParityDbWeight: RuntimeDbWeight = RuntimeDbWeight {
+			read: 8_000 * WEIGHT_REF_TIME_PER_NANOS,
+			write: 50_000 * WEIGHT_REF_TIME_PER_NANOS,
+		};
+
+		/// Importing a block with 0 Extrinsics.
+		pub const BlockExecutionWeight: Weight =
+			Weight::from_parts(WEIGHT_REF_TIME_PER_NANOS.saturating_mul(5_000_000), 0);
+		/// Executing a NO-OP `System::remarks` Extrinsic.
+		pub const ExtrinsicBaseWeight: Weight =
+			Weight::from_parts(WEIGHT_REF_TIME_PER_NANOS.saturating_mul(125_000), 0);
```

Comment:
Across all the testnet system parachains they're configured to the exact same value. I was surprised at this, as I expected each runtime to be different in some way. There's no benchmark or anything that sets this directly IIUC so I thought it could be lifted out to a common definition. If any single runtime wants to override it, they can do it easily by just specifying it.
Maybe I'm being greedy, but anything that's configured the same I've taken away.

Reply:
For these two for example:
```bash
polkadot-sdk on  donal-SP-default-configs [$] via 🦀 v1.75.0 
❯ rg "const BlockExecutionWeight" -A1 cumulus
cumulus/parachains/runtimes/testing/penpal/src/weights/block_weights.rs
26:     pub const BlockExecutionWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(5_000_000), 0);

cumulus/parachains/runtimes/coretime/coretime-westend/src/weights/block_weights.rs
26:     pub const BlockExecutionWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(5_000_000), 0);

cumulus/parachains/runtimes/assets/asset-hub-rococo/src/weights/block_weights.rs
26:     pub const BlockExecutionWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(5_000_000), 0);

cumulus/parachains/runtimes/constants/src/lib.rs
54:     pub const BlockExecutionWeight: Weight =
55-         Weight::from_parts(WEIGHT_REF_TIME_PER_NANOS.saturating_mul(5_000_000), 0);

cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/weights/block_weights.rs
26:     pub const BlockExecutionWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(5_000_000), 0);

cumulus/parachains/runtimes/contracts/contracts-rococo/src/weights/block_weights.rs
26:     pub const BlockExecutionWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(5_000_000), 0);

cumulus/parachains/runtimes/bridge-hubs/bridge-hub-westend/src/weights/block_weights.rs
26:     pub const BlockExecutionWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(5_000_000), 0);

cumulus/parachains/runtimes/assets/asset-hub-westend/src/weights/block_weights.rs
26:     pub const BlockExecutionWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(5_000_000), 0);

cumulus/parachains/runtimes/people/people-westend/src/weights/block_weights.rs
26:     pub const BlockExecutionWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(5_000_000), 0);

cumulus/parachains/runtimes/people/people-rococo/src/weights/block_weights.rs
26:     pub const BlockExecutionWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(5_000_000), 0);

cumulus/parachains/runtimes/collectives/collectives-westend/src/weights/block_weights.rs
26:     pub const BlockExecutionWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(5_000_000), 0);

polkadot-sdk on  donal-SP-default-configs [$] via 🦀 v1.75.0 
❯ rg "const ExtrinsicBaseWeight" -A1 cumulus
cumulus/parachains/runtimes/assets/asset-hub-rococo/src/weights/extrinsic_weights.rs
26:     pub const ExtrinsicBaseWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(125_000), 0);

cumulus/parachains/runtimes/people/people-westend/src/weights/extrinsic_weights.rs
26:     pub const ExtrinsicBaseWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(125_000), 0);

cumulus/parachains/runtimes/coretime/coretime-westend/src/weights/extrinsic_weights.rs
26:     pub const ExtrinsicBaseWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(125_000), 0);

cumulus/parachains/runtimes/assets/asset-hub-westend/src/weights/extrinsic_weights.rs
26:     pub const ExtrinsicBaseWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(125_000), 0);

cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/weights/extrinsic_weights.rs
26:     pub const ExtrinsicBaseWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(125_000), 0);

cumulus/parachains/runtimes/people/people-rococo/src/weights/extrinsic_weights.rs
26:     pub const ExtrinsicBaseWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(125_000), 0);

cumulus/parachains/runtimes/testing/penpal/src/weights/extrinsic_weights.rs
26:     pub const ExtrinsicBaseWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(125_000), 0);

cumulus/parachains/runtimes/contracts/contracts-rococo/src/weights/extrinsic_weights.rs
26:     pub const ExtrinsicBaseWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(125_000), 0);

cumulus/parachains/runtimes/collectives/collectives-westend/src/weights/extrinsic_weights.rs
26:     pub const ExtrinsicBaseWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(125_000), 0);

cumulus/parachains/runtimes/constants/src/lib.rs
57:     pub const ExtrinsicBaseWeight: Weight =
58-         Weight::from_parts(WEIGHT_REF_TIME_PER_NANOS.saturating_mul(125_000), 0);

cumulus/parachains/runtimes/bridge-hubs/bridge-hub-westend/src/weights/extrinsic_weights.rs
26:     pub const ExtrinsicBaseWeight: Weight =
27-         Weight::from_parts(constants::WEIGHT_REF_TIME_PER_NANOS.saturating_mul(125_000), 0);
```

Reply:
@ggwpez there exists some benchmark for this or not?

Reply:
Yea these two should not be hardcoded but benchmarked. The fact that we dont do that is a bug rather than a feature.